pkg/cmd/install.go

```diff
@@ -121,15 +131,61 @@ func expandPath(path string) string {
 	return res
 }
 
-func loadPublickey(path string) ssh.AuthMethod {
+func sshAgent(publicKeyPath string) ssh.AuthMethod {
+	if sshAgentConn, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK")); err == nil {
```

Comment:
`defer sshAgentConn.Close()`?

Reply:
The sshAgent (sshAgent := agent.NewClient(sshAgentConn)) wouldn't work anymore if i close the underlying connection to the ssh-agent.

Reply:
Hmm.. could we leave a dangling connection?

Reply:
Hm, I think we are fine because k3sup is just a short living program.
I have nevertheless adapted the code so that the connection is now always closed.